trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/credentials/http/TestHttpCredentialsProviderServer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.credentials.http;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.http.server.HttpServerConfig;
+import io.airlift.http.server.HttpServerInfo;
+import io.airlift.http.server.testing.TestingHttpServer;
+import io.airlift.node.NodeInfo;
+import io.trino.aws.proxy.server.testing.TestingIdentity;
+import io.trino.aws.proxy.server.testing.TestingTrinoAwsProxyServer;
+import io.trino.aws.proxy.server.testing.harness.BuilderFilter;
+import jakarta.servlet.Servlet;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static io.trino.aws.proxy.server.credentials.http.HttpCredentialsModule.HTTP_CREDENTIALS_PROVIDER_IDENTIFIER;
+import static io.trino.aws.proxy.spi.plugin.TrinoAwsProxyServerBinding.bindIdentityType;
+
+public class TestHttpCredentialsProviderServer
+{
+    protected TestHttpCredentialsProviderServer() {}
+
+    public static class Filter
+            implements BuilderFilter
+    {
+        private static String httpEndpointUri;
+
+        @Override
+        public TestingTrinoAwsProxyServer.Builder filter(TestingTrinoAwsProxyServer.Builder builder)
```

Comment:
This filter, unlike the TestingHttpCredentialsServlet, would be of limited use to anything except for a single test class (where it was moved from), I think that it should be moved back under `TestHttpCredentialsProvider`